Samples/MapScenarios/MapScenarios/Scenarios/OffRoute/OffRouteViewController.swift

```diff
@@ -0,0 +1,198 @@
+                }
+            })
+        }
+    }
+
+    func startManagedLocationManager() {
+        DispatchQueue.main.async { [weak self] in
+            guard let buildingIdentifier = self?.buildingIdentifier else {
+                return
+            }
+            let managedLocationManager = PWManagedLocationManager(buildingId: buildingIdentifier)
+            self?.mapView.register(managedLocationManager)
+        }
+    }
+
+    func configureMapViewConstraints() {
+        mapView.translatesAutoresizingMaskIntoConstraints = false
+        mapView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
+        mapView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        mapView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+        mapView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
+    }
+
+    func buildRoute() {
+        dontShowAgain = false
+
+        var destinationPOI: PWPointOfInterest!
+        if destinationPOIIdentifier != 0 {
+            destinationPOI = mapView.building.pois.filter({
+                return $0.identifier == destinationPOIIdentifier
+            }).first
+        } else {
+            if let firstPOI = mapView.building.pois.first {
+                destinationPOI = firstPOI
+            }
+        }
+
+        if destinationPOI == nil {
+            print("No points of interest found, please add at least one to the building in the Maas portal")
+            return
+        }
+
+        PWRoute.createRoute(from: mapView.indoorUserLocation, to: destinationPOI, accessibility: false, excludedPoints: nil, completion: { [weak self] (route, error) in
+            if (route != nil) {
+                self?.currentRoute = route
+            } else {
+                print("Couldn't find a route from you current location to the destination.")
+                return
+            }
+
+            let routeOptions = PWRouteUIOptions()
+            self?.mapView.navigate(with: route, options: routeOptions)
+        })
+    }
+
+    @objc func fireTimer() {
+        offRouteTimer?.invalidate()
+        offRouteTimer = nil
+        showModal()
+    }
+
+    private func showModal() {
+        if (!modalVisible) {
+            modalVisible = true
+
+            let offRouteModal = OffRouteModalViewController()
+            offRouteModal.modalPresentationStyle = .overCurrentContext
+            offRouteModal.modalTransitionStyle = .crossDissolve
+
+            offRouteModal.dismissCompletion = {
+                print("JFULLEN - dismiss completion handler")
+                self.modalVisible = false
+            }
+
+            offRouteModal.rerouteCompletion = {
+                print("JFULLEN - reroute completion handler")
+                self.modalVisible = false
+                self.mapView.cancelRouting()
+                self.currentRoute = nil
+                self.buildRoute()
```

Comment:
could we use [weak self] here, or is there a strong reason not to? it seems like this would probably dismiss okay since the presenting VC doesn't actually own the modal but if there was a reference added from presenter to modal you could have a retain cycle here

Reply:
Done